packages/react/src/components/Modal/Modal.tsx

```diff
@@ -564,7 +564,7 @@ const Modal = React.forwardRef(function Modal(
       open={open}
       modal
       ref={innerModal}
-      role={isAlertDialog ? 'alertdialog' : ''}
+      role={isAlertDialog ? 'alertdialog' : 'dialog'}
```

Comment:
The spec recommends against Dialog having `role='dialog'` https://www.w3.org/TR/html-aria/#docconformance:~:text=Role%3A-,alertdialog,-.%20(dialog%20is

```suggestion
      role={isAlertDialog ? 'alertdialog' : ''}
```
I put that in the comments up above this a few lines where isAlertDialog is defined. Maybe it would be better to fully encapsulate that prop so there's no confusion.

```js
// Dialog elements can only have an `aria-role` of `alertdialog` or `dialog`, and `dialog` is not recommended, so only `alertdialog` is used here.
// https://www.w3.org/TR/html-aria/#docconformance:~:text=Role%3A-,alertdialog,-.%20(dialog%20is
const role = alert && !passiveModal ? 'alertdialog' : '';

...and then down here at the usage
<Dialog
  ...
  role={role}
```